### DISCUSSION TITLE
CollectionChanged during SaveAndMergeAsync

### DISCUSSION THREAD

**Discussion by lwmorris067** [MarimerLLC/csla#4169] [Questions]
Just curious, do business rules run when the Merge portion of the SaveAndMergeAsync executes? I was having some really odd issues in my integration tests. I changed things over to SaveAsync and everythings works, but SaveAndMergeAsync gives me a CollectionChanged error from within CSLA.

Here what I get:

 Csi.Waiver.Ui.Shared.IntegrationTests.Model.Case.Claims.Rules.ComputeAgeDisabledTests.Verify_Rule_Runs_After_Fetch
   Source: ComputeAgeDisabledTests.cs line 16
   Duration: 2.9 sec

  Message: 
System.InvalidOperationException : Collection was modified; enumeration operation may not execute.

  Stack Trace: 
Enumerator.MoveNextRare()
GraphMerger.MergeBusinessListGraph[T,C](T target, T source)
RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr)

**Comment by StefanOssendorf** [comment]
Hi. The Merger uses `LoadProperty` to set values. So that should not raise any events. Could you please create a minimal working example to get that error and we can look into it?
Edit: That's not entirly correct. For collection changes the appropiate events are invoked because the merger adds/deletes to the list like user code would. So doint something in a property/collection changed handler of that list could introduce that problem.
